src/panel/pages/timeline/components/TimelineEvent.tsx

```diff
@@ -14,25 +18,52 @@ const getEventType = (event: ParsedEvent) => {
   return "addition";
 };
 
-export const TimelineEvent = styled.div.attrs<{ event: ParsedEvent }>(
-  props => ({
-    "data-type": getEventType(props.event)
-  })
-)<{ event: ParsedEvent }>`
+const COLOR_MAPPING = {
+  addition: importedTheme.green["0"],
+  update: importedTheme.orange["0"],
+  teardown: importedTheme.grey["-1"]
+};
+
+const Wrapper = styled.div`
+  padding: 50px;
+`;
+
+const Container = styled.div`
   border-radius: 50%;
   width: 10px;
   height: 10px;
   border: solid 2px ${props => props.theme.dark["+1"]};
+  background: ${props => props.color};
+`;
 
-  &[data-type="addition"] {
-    background: ${props => props.theme.green["0"]};
-  }
+const getPosFromRef = (
```

Comment:
This is exactly what I was thinking - good shout 🔥 🔥 

You okay with us moving this to a hook / component?